app/helpers/admin/firms_helper.rb

```diff
@@ -16,6 +16,10 @@ def render_questionnaire_response(firm, attribute_name)
     end
   end
 
+  def user_for(principal)
+    User.find_by_principal_token principal.token
+  end
```

Comment:
Ah I understand what you were talking about earlier now — principal has no method on it to return the user.